CMakeLists.txt

```
@@ -1,16 +1,58 @@
 cmake_minimum_required (VERSION 3.3)
 project (llama)
 
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_STANDARD 11)
+# llama
+find_package(Boost REQUIRED)
+add_library(${PROJECT_NAME} INTERFACE)
+target_include_directories(${PROJECT_NAME} INTERFACE $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include> $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+target_compile_features(${PROJECT_NAME} INTERFACE cxx_std_11)
+target_link_libraries(${PROJECT_NAME} INTERFACE Boost::headers)
 
-ADD_SUBDIRECTORY("examples")
+# llama::llama to make subdirectory projects work
+add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
 
+# llama IDE
+file(GLOB_RECURSE llamaSources "${CMAKE_CURRENT_SOURCE_DIR}/include/**")
+add_custom_target("llamaIde" SOURCES ${llamaSources})
+source_group(TREE "${CMAKE_CURRENT_LIST_DIR}/include/llama" FILES ${llamaSources})
+
+# tests
+find_package(Catch2 CONFIG REQUIRED)
```

Maybe at least as indicator:
```suggestion
find_package(Catch2 2.6.1 CONFIG REQUIRED)
```